scene/3d/mesh_instance_3d.cpp

```diff
@@ -493,6 +493,93 @@ void MeshInstance3D::create_debug_tangents() {
+			}
+			new_mesh_arrays[i] = source_mesh_arrays[i];
+		}
+
+		Vector<Vector3> lerped_vertex_array = source_mesh_vertex_array;
+		Vector<Vector3> lerped_normal_array = source_mesh_normal_array;
+		Vector<float> lerped_tangent_array = source_mesh_tangent_array;
+
+		const Array &blendshapes_mesh_arrays = source_mesh->surface_get_blend_shape_arrays(surface_index);
+		int blend_shape_count = source_mesh->get_blend_shape_count();
+		ERR_FAIL_COND_V(blendshapes_mesh_arrays.size() != blend_shape_count, new_baked_mesh);
+
+		for (int blendshape_index = 0; blendshape_index < blend_shape_count; blendshape_index++) {
+			float blend_weight = get_blend_shape_value(blendshape_index);
+			if (abs(blend_weight) > 0.0001) {
+				const Array &blendshape_mesh_arrays = blendshapes_mesh_arrays[blendshape_index];
+
+				const Vector<Vector3> &blendshape_vertex_array = blendshape_mesh_arrays[Mesh::ARRAY_VERTEX];
+				const Vector<Vector3> &blendshape_normal_array = blendshape_mesh_arrays[Mesh::ARRAY_NORMAL];
+				const Vector<float> &blendshape_tangent_array = blendshape_mesh_arrays[Mesh::ARRAY_TANGENT];
+
+				ERR_FAIL_COND_V(source_mesh_vertex_array.size() != blendshape_vertex_array.size(), new_baked_mesh);
+				ERR_FAIL_COND_V(source_mesh_normal_array.size() != blendshape_normal_array.size(), new_baked_mesh);
+				ERR_FAIL_COND_V(source_mesh_tangent_array.size() != blendshape_tangent_array.size(), new_baked_mesh);
+
+				for (int i = 0; i < source_mesh_vertex_array.size(); i++) {
+					const Vector3 &source_vertex = source_mesh_vertex_array[i];
+					const Vector3 &blendshape_vertex = blendshape_vertex_array[i];
+					Vector3 lerped_vertex = source_vertex.lerp(blendshape_vertex, blend_weight) - source_vertex;
+					lerped_vertex_array.write[i] += lerped_vertex;
+
+					const Vector3 &source_normal = source_mesh_normal_array[i];
+					const Vector3 &blendshape_normal = blendshape_normal_array[i];
+					Vector3 lerped_normal = source_normal.lerp(blendshape_normal, blend_weight) - source_normal;
+					lerped_normal_array.write[i] += lerped_normal;
+
+					int tangent_index = i * 4;
+					const Vector4 source_tangent = Vector4(
+							source_mesh_tangent_array[tangent_index],
+							source_mesh_tangent_array[tangent_index + 1],
+							source_mesh_tangent_array[tangent_index + 2],
+							source_mesh_tangent_array[tangent_index + 3]);
+					const Vector4 blendshape_tangent = Vector4(
+							blendshape_tangent_array[tangent_index],
+							blendshape_tangent_array[tangent_index + 1],
+							blendshape_tangent_array[tangent_index + 2],
+							blendshape_tangent_array[tangent_index + 3]);
+					Vector4 lerped_tangent = source_tangent.lerp(blendshape_tangent, blend_weight);
+					lerped_tangent_array.write[tangent_index] += lerped_tangent.x;
+					lerped_tangent_array.write[tangent_index + 1] += lerped_tangent.y;
+					lerped_tangent_array.write[tangent_index + 2] += lerped_tangent.z;
+					lerped_tangent_array.write[tangent_index + 3] += lerped_tangent.w;
```

Comment:
Meshes are allowed to not have normals or tangents. The previous `ERR_FAIL_COND` correctly addresses this by checking if the sizes match (that is to say, if they do not have normals or tangents, the blend shapes should not have normals or tangents.

However, the highlighted code in the inner for loop will access out of bounds in the case that the arrays are empty (or otherwise do not match the size of vertex_array).

these should only run `if source_mesh_normal_array.size() == source_mesh_vertex_array.size()` and `if source_mesh_tangent_array.size() / 4 == source_mesh_vertex_array.size()`